### DISCUSSION TITLE
Can we get `response` from `WebSocketSession` ?

### DISCUSSION THREAD

**Discussion by WSH032** [frankie567/httpx-ws#53] [Ideas]
https://github.com/frankie567/httpx-ws/blob/1e1c252c2b678f8cc475e1c1546980a784f19702/httpx_ws/_api.py#L111

After successfully establishing a WebSocket connection, we can get the server's accepted subprotocol from `ws.subprotocol`.

**But, how about adding a method to get the response?**

---

It could be beneficial.

Refer to [starlette/websockets/#accepting-the-connection](https://www.starlette.io/websockets/#accepting-the-connection), there is a `headers` to response

```python
await websocket.accept(subprotocol=None, headers=None)
```

If we could get `response headers` from the `WebSocketSession`, perhaps we could implement a transparent proxy server.

Somethin like

```python
ws: AsyncWebSocketSession = ...
await websocket.accept(subprotocol=ws.subprotocol, headers=ws.response.headers)
```

---

How about:
```python
def __init__(
    self
    ...,
    respose: httpx.Response,
):
    self.response = response
```

**Comment by frankie567** [comment]
Interesting, yes! Having the whole original `Response` object could be useful in some contexts. PR welcome!

**Reply by WSH032** [reply]
Thanks! PR: #54
